File: Python/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        i = 0
        for num in nums:
            i = (i + 1 if num else 0)
            res = max(res, i)
        return res

File: Swift/max-consecutive-ones.swift
class Solution {
    func findMaxConsecutiveOnes(_ nums: [Int]) -> Int {
        var res = 0
        var i = 0

        for num in nums {
            i = num == 1 ? i + 1 : 0
            res = max(res, i)
        }

        return res
    }
}
